softcut-lib/include/softcut/Resampler.h

```diff
@@ -132,15 +164,42 @@ namespace softcut {
             phase_t f = 1.0 - phase_;
             // normalized (divided by rate);
             f = f * phi_;
-            // write the first output frame
             unsigned int i=0;
-            outBuf_[i] = interpolate(f);
-            while(i < nf) {
-                // distance between output frames in this normalized space is 1/rate
-                f += phi_;
-                outBuf_[i] = interpolate(f);
-                i++;
+
+            switch(mode_) {
```

Comment:
please do not add a switch statement here. there is a standard way to avoid it and do what you are trying to do in C++ DSP